addon/components/polaris-text-container.js

```diff
@@ -1,53 +1,43 @@
 import Component from '@ember/component';
 import { computed } from '@ember/object';
 import { classify } from '@ember/string';
+import { tagName, layout as templateLayout } from '@ember-decorators/component';
 import layout from '../templates/components/polaris-text-container';
 
 const allowedSpacings = ['tight', 'loose'];
 
 /**
  * Undocumented Polaris text container component.
  */
-export default Component.extend({
-  classNames: ['Polaris-TextContainer'],
-
-  classNameBindings: ['spacingClass'],
-
-  layout,
-
+@tagName('')
+@templateLayout(layout)
+export default class PolarisTextContainer extends Component {
   /**
    * The text to display.
    *
    * This component can be used in block form,
    * in which case the block content will be used
    * instead of `text`
    *
-   * @property text
-   * @type {string}
-   * @default null
+   * @type {String}
    * @public
    */
-  text: null,
+  text;
 
   /**
    * The amount of vertical spacing children will get between them.
    *
-   * @property spacing
-   * @type {string}
-   * @default null
+   * @type {String}
    * @public
    */
-  spacing: null,
+  spacing;
 
-  /**
-   * @private
-   */
-  spacingClass: computed('spacing', function() {
-    let spacing = this.get('spacing');
-    if (allowedSpacings.indexOf(spacing) > -1) {
-      return `Polaris-TextContainer--spacing${classify(spacing)}`;
+  @(computed('spacing').readOnly())
+  get spacingClass() {
```

Comment:
we don't need to mark this as `readOnly` since we've only defined the getter.

Reply:
💯